transformations/mixed_language_perturbation/transformation.py

```diff
@@ -0,0 +1,93 @@
+import random
+import string
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+from transformers import M2M100ForConditionalGeneration, M2M100Tokenizer
+
+"""
+Mixed Language Perturbation
+This perturbation translates words in the text from English to other languages (e.g., German). It can be used to test the robustness of a model in a multilingual setting.
+"""
+
+def translate(model, tokenizer, text, src_lang, target_lang):
+    tokenizer.src_lang = src_lang
+    encoded = tokenizer(text, return_tensors="pt")
+    generated_tokens = model.generate(**encoded, forced_bos_token_id=tokenizer.get_lang_id(target_lang))
+    res = tokenizer.batch_decode(generated_tokens, skip_special_tokens=True)
+    return res
+
+def mixed_language(model, tokenizer, text, prob_mix=0.1, src_lang="en", trg_lang="fr", seed=0):
+    random.seed(seed)
+
+    words = text.split()
+    mixed_text = ""
+    for word in words:
+        if mixed_text != "":
+            mixed_text += " "
+        if random.random() < prob_mix:
```

Comment:
In the paper you cited, you seem to select the word with the highest attention score rather than at random. Is there any performance difference on cross-lingual training between random and highest attention?
Even if you select words at random, that's fine as German words in this case can probably work as regularizers - but why not use a bilingual lexicon - since you are anyways translating words?

Reply:
@kaustubhdhole : We choose random instead of taking the highest attention score, as it is a much cheaper approach and can produce more diverse augmentation. And you are also correct that it can work as a regularizer as well. We don't use a bilingual lexicon as it will produce a large multilingual dictionary, and the replacement is limited to the word inside the dictionary.

We think it is another good idea to create a light (`heavy=False`) version of this perturbation with a bilingual lexicon. Should we add another perturbation for this matter or combine both approaches using lexicon for common words and model translation for uncommon words?

Reply:
Thanks for the clarification. That'd be a great idea! You can add it as a separate perturbation/transformation from the current one.